### PR TITLE
Route tracker

### PR DESCRIPTION
**Pull request by dyfero**

### CODE REVIEW

File: src/Trackers/RouteTracker.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    public function checkRequest(Request $request): bool
+    {
+        $path = $request->getPathInfo();
+        return $this->isEnabled() && $this->hasPrefix($path) && !$this->isInIgnoreUri($path);
+    }
+
+    public function trackRoute(Request $request): void
+    {
+        $this->repository->create([
+            'user_id' => auth()->user() ? auth()->user()->id : null,
```
Comment (HerbertIV) [inline src/Trackers/RouteTracker.php:49]:
```suggestion
            'user_id' => auth()->user()->id ?? null,
```

File: database/migrations/2022_03_02_145348_create_track_routes_table.php [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('track_routes', function (Blueprint $table) {
+            $table->id();
+            $table->unsignedBigInteger('user_id')->nullable();
```
Comment (HerbertIV) [inline database/migrations/2022_03_02_145348_create_track_routes_table.php:18]: Why do not maked a foreign key?

**Review by HerbertIV — APPROVED** [review]

File: database/migrations/2022_03_02_145348_create_track_routes_table.php (hunk 2 above)
Comment (qunabu) [reply database/migrations/2022_03_02_145348_create_track_routes_table.php:18]:
```suggestion
                $table->foreign('user_id')->references('id')->on('users');
```

seems thgat casade delete is missing https://laravel.com/docs/9.x/migrations#foreign-key-constraints 

```php
$table->foreignId('user_id')
      ->constrained()
      ->onUpdate('cascade')
      ->onDelete('cascade');
```

**Review by qunabu — CHANGES_REQUESTED** [review]

File: src/Dto/TrackRouteSearchDto.php [hunk 3]
```diff
@@ -0,0 +1,58 @@
+<?php
+
+namespace EscolaLms\Tracker\Dto;
+
+use EscolaLms\Core\Dtos\Contracts\DtoContract;
+use EscolaLms\Core\Dtos\Contracts\InstantiateFromRequest;
+use Illuminate\Http\Request;
+
+class TrackRouteSearchDto implements DtoContract, InstantiateFromRequest
```
Comment (qunabu) [inline src/Dto/TrackRouteSearchDto.php:11]: `dateFrom` and `dateTo` is essential

**Review by qunabu — CHANGES_REQUESTED** [review]
1. keys on user 
2. search by date range

**Pull request merged**